.github/workflows/zephyr.yml

```diff
@@ -70,3 +69,135 @@ jobs:
+          main,
+        ]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          path: ./workspace/sof
+
+      # Cache artifacts so we do not overload external servers with downloads
+      # Remember to change step key if you change the tools so old cache is not restored,
+      # or delete cache manually in Github Actions tab
+      - name: Cache unzip
+        id: cache-unzip
+        uses: actions/cache@v3.1.0-beta.3
+        with:
+          path: unzip-5.51-1-bin.zip
+          key: ${{ runner.os }}-cache-unzip-5-51-1
+
+      - name: Cache wget
+        id: cache-wget
+        uses: actions/cache@v3.1.0-beta.3
+        with:
+          path: wget-1.11.4-1-bin.zip
+          key: ${{ runner.os }}-cache-wget-1-11-4-1
+
+      - name: Cache Zephyr SDK
+        id: cache-zephyr-sdk
+        uses: actions/cache@v3.1.0-beta.3
+        with:
+          path: zephyr-sdk-0.15.2_windows-x86_64.zip
+          key: ${{ runner.os }}-cache-zephyr-sdk-0-15-2
+
+      # Try restoring needed artifacts from Cache
+      - if: ${{ steps.cache-unzip.outputs.cache-hit != 'true' }}
+        name: Download unzip
+        run: |
+          curl -L -o unzip-5.51-1-bin.zip `
+          https://gnuwin32.sourceforge.net/downlinks/unzip-bin-zip.php
+
+      - if: ${{ steps.cache-wget.outputs.cache-hit != 'true' }}
+        name: Download wget
+        run: |
+          curl -L -O http://downloads.sourceforge.net/gnuwin32/wget-1.11.4-1-bin.zip
+
+      - if: ${{ steps.cache-zephyr-sdk.outputs.cache-hit != 'true' }}
+        name: Download Zephyr SDK
+        run: |  # yamllint disable-line rule:line-length
+          curl -L -O `
+          https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v0.15.2/zephyr-sdk-0.15.2_windows-x86_64.zip
+
+      # Unzips every .zip package to directory matching its name without extension
+      - name: Unzip downloaded packages
+        run: 7z x *.zip -o*
+
+      - name: Add unzip to system PATH
+        run: |
+          echo "${{ github.workspace }}\unzip-5.51-1-bin\bin" | `
+          Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+
+      - name: Add wget to system PATH
+        run: |
+          echo "${{ github.workspace }}\wget-1.11.4-1-bin\bin" | `
+          Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+
+      # Install Zephyr SDK - all toolchains including Host Tools
+      # and registering CMake package in the registry
+      - name: Install Zephyr SDK
+        run: zephyr-sdk-0.15.2_windows-x86_64\zephyr-sdk-0.15.2\setup.cmd /t all /h /c
+
+      # Cache dependency path is always relative to ${{ github.workspace }}
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+          cache: 'pip'
+          cache-dependency-path: "**\requirements*.txt"
+
+      - name: West install
+        run: pip3 install west
+
+      - if: ${{ matrix.zephyr_revision }} == 'main'
```

Comment:
Maybe drop this to simplify and reduce the duplication with Linux. Building the manifest version should be enough with Windows, Linux should be enough to find upstream regressions. This will also reduce the number of "checks" in the too small window, it's getting too big.